ruby/lib/ci/queue/redis/base.rb

```diff
@@ -7,8 +7,8 @@ def initialize(redis:, build_id:)
           @build_id = build_id
         end
 
-        def empty?
-          size == 0
+        def exhausted?
```

Comment:
I renamed `empty?` to `exhausted?` to more clearly reflect that the queue is empty because it was fully worked off, and not because it's not populated yet.